lnwallet/test/test_interface.go

```diff
@@ -1819,16 +1819,27 @@ func testPublishTransaction(r *rpctest.Harness,
 			require.NoError(t, err, "unable to obtain public key")
 
 			// Create a new transaction that spends the output from
-			// tx3, and that pays to a different address. We expect
-			// this to be rejected because it is a double spend.
+			// tx3, and that pays to a different address.
 			tx5, err := txFromOutput(
 				tx3, alice.Cfg.Signer, keyDesc.PubKey,
 				keyDesc2.PubKey, txFee, rbf,
 			)
 			require.NoError(t, err)
 
 			err = alice.PublishTransaction(tx5, labels.External)
-			require.ErrorIs(t, err, lnwallet.ErrDoubleSpend)
+
+			// If RBF is not enabled, we expect this to be rejected
+			// because it is a double spend.
+			expectedErr := lnwallet.ErrDoubleSpend
+
+			// If RBF is enabled, we expect it to be rejected
+			// because it doesn't pay enough fees.
+			if rbf {
+				expectedErr = rpcclient.ErrInsufficientFee
```

Comment:
Q: wondering whether we should get rid of the internal LND mapping or add this error to the mapping ?

Reply:
I'd like to get rid of the internal LND mapping and expose the detailed RPC errors to the callers - we cannot do it yet tho, as I think there would be compatibility issues.